File: src/main/java/com/wooteco/wiki/repository/DocumentRepository.java
package com.wooteco.wiki.repository;

import com.wooteco.wiki.entity.Document;
import com.wooteco.wiki.entity.Title;
import java.util.List;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface DocumentRepository extends JpaRepository<Document, Long> {
    Optional<Document> findByTitle(String title);

    boolean existsByTitle(String title);

    List<Document> findAllByOrderByGenerateTimeDesc();

    List<Title> findAllByTitleStartingWith(String keyWord);
}

File: src/main/java/com/wooteco/wiki/repository/DocumentRepository.kt
package com.wooteco.wiki.repository

import com.wooteco.wiki.domain.Document
import com.wooteco.wiki.domain.Title
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface DocumentRepository : JpaRepository<Document, Long> {
    fun findByTitle(title: String): Optional<Document>

    fun existsByTitle(title: String): Boolean

    fun findAllByOrderByGenerateTimeDesc(): List<Document>

    fun findAllByTitleStartingWith(keyWord: String): List<Title>

    fun findByUuid(uuid: UUID): Optional<Document>
}
